pkg/client/kubernetes/chartapplier_test.go

```diff
@@ -92,144 +93,164 @@ var _ = Describe("chart applier", func() {
+					ctx,
+					embeddedFS,
+					chartPath,
+					newNS,
+					name,
+					kubernetes.Values(val),
+					kubernetes.ForceNamespace,
+					nil, // simulate nil entry
+					m,
+					nil, // simulate nil entry
+				)).To(Succeed())
+
+				actual := &corev1.ConfigMap{}
+				Expect(c.Get(ctx, client.ObjectKey{Name: configMapName, Namespace: newNS}, actual)).To(Succeed())
+				Expect(actual).To(DeepDerivativeEqual(expectedCM))
+			})
+		}
+
+		Context("chart apiVersion: v1", func() {
+			test(chartPathV1)
 		})
 
-		It("renders the chart with custom values", func() {
-			const newNS = "other-namespace"
-
-			existingCM := &corev1.ConfigMap{
-				TypeMeta: configMapTypeMeta,
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      configMapName,
-					Namespace: newNS,
-				},
-			}
-			Expect(c.Create(ctx, existingCM)).To(Succeed(), "dummy configmap creation should succeed")
-
-			expectedCM.Namespace = newNS
-			expectedCM.Data = map[string]string{"key": "new-value"}
-			expectedCM.Labels = map[string]string{"baz": "new-foo"}
-			expectedCM.Annotations = map[string]string{"new": "new-value"}
-			expectedCM.ResourceVersion = "2"
-
-			val := &renderTestValues{
-				Service: renderTestValuesService{
-					Data:   map[string]string{"key": "new-value"},
-					Labels: map[string]string{"baz": "new-foo"},
-				},
-			}
-
-			m := kubernetes.MergeFuncs{
-				corev1.SchemeGroupVersion.WithKind("ConfigMap").GroupKind(): func(newObj, oldObj *unstructured.Unstructured) {
-					newObj.SetAnnotations(map[string]string{"new": "new-value"})
-				},
-			}
-
-			Expect(ca.ApplyFromEmbeddedFS(
-				ctx,
-				embeddedFS,
-				chartPath,
-				newNS,
-				name,
-				kubernetes.Values(val),
-				kubernetes.ForceNamespace,
-				nil, // simulate nil entry
-				m,
-				nil, // simulate nil entry
-			)).To(Succeed())
-
-			actual := &corev1.ConfigMap{}
-			Expect(c.Get(ctx, client.ObjectKey{Name: configMapName, Namespace: newNS}, actual)).To(Succeed())
-			Expect(actual).To(DeepDerivativeEqual(expectedCM))
+		Context("chart apiVersion: v2", func() {
+			test(chartPathV2)
 		})
 	})
 
 	Describe("#DeleteFromEmbeddedFS()", func() {
-		It("deletes the chart with default values", func() {
-			existingCM := &corev1.ConfigMap{
-				TypeMeta: configMapTypeMeta,
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      configMapName,
-					Namespace: configMapName,
-				},
-			}
-			Expect(c.Create(ctx, existingCM)).To(Succeed(), "dummy configmap creation should succeed")
-
-			Expect(ca.DeleteFromEmbeddedFS(ctx, embeddedFS, chartPath, namespace, name)).To(Succeed())
-
-			Expect(c.Get(ctx, client.ObjectKey{Name: configMapName, Namespace: namespace}, &corev1.ConfigMap{})).To(BeNotFoundError())
-		})
+		test := func(chartPath string) {
+			It("deletes the chart with default values", func() {
+				existingCM := &corev1.ConfigMap{
+					TypeMeta: configMapTypeMeta,
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      configMapName,
+						Namespace: configMapName,
```

Comment:
Not introduced with this PR, but there is small bug in this test
```suggestion
						Namespace: namespace,
```

(Creates a ConfigMap in ns "test-configmap-name", then deletes the chart from ns "test-chart-namespace" and verifies the config map is missing from ns "test-chart-namespace").